File: client-app/src/app/models/employee.ts
import { EmployeeHrFormValues, IEmployeeHr } from "./employeeHr";
import { EmployeeItFormValues, IEmployeeIt } from "./employeeIt";
import { EmployeeTrainingFormValues, IEmployeeTraining } from "./employeeTraining";

export interface IEmployeeEnvelope {
    employees: IEmployee[];
    employeeCount: number;
}

export interface IEmployee {

    adminAssistant: string;
    billingTitle: string;
    cellPhone: string;
    clientName: string;
    comments: string;
    companyCode: string;
    computer: string;
    //Date: Date;

    typeTechnical: string;  // 1 = non-technical, 2 = technical, 3 = row

    didNotAcceptCbx: boolean;
    
    typeCbxUnion: boolean;
    typeCbxBManager: boolean;
    typeCbxNewHire: boolean;
    typeCbxReHire: boolean;
    typeCbxETChange: boolean;
    typeCbxTransfer: boolean;

    gotCbxMNFStandards: boolean;
    gotCbxTEExpenses: boolean;
    gotCbxMInternet: boolean;

    firstName: string;
    lastName: string;
    number: string;         // employee number

    startDate?: Date;
    startTime?: Date;

    office: string;
    location: string;
    pwLocation: string;
    employeeType: string;

    discipline: string;
    //discipline?: number;

    sF330Discipline: string;
    //sF330Discipline?: number;
    jobTitle: string;

    designation: string;
    //designation: number;

    managerName: string;
    managerEmail: string;

    typeCbxMPosition: boolean;
    typeCbxESAMTraining: boolean;



    ssNeeded: string;
    shNeeded: string;
    seatingArrangement: string;

    cvCbxEDMVehicle: boolean;
    cvCbxEAMVehicle: boolean;
    cvCbxGasCard: boolean;
    cvCbxEZPass: boolean;

    e52Week: string;
    //e52Week?: number;
    e48Week: string;
    //e48Week?: number;

    inCbxBCards: boolean;
    inCbxSPhone: boolean;
    inCbxPResume: boolean;
    inCbxLaptop: boolean;           // Office Laptop
    inCbxFieldLaptop: boolean;
    inCbxDesktop: boolean;
    inCbxUsbAccess: boolean;
    inCbxDPhone: boolean;
    inCbxOther: boolean;
    inTxtOther: string;

    inCbxMeetWithMarketing: boolean;

    seCbxHHats: boolean;
    seCbxSVests: boolean;
    seCbxSGlasses: boolean;
    seCbxOther: boolean;

    seHatColor: string;
    seNeededType: string;
    seSize: string;
    seTxtOther: string;

    //employeeNumber: string;
    employeeTraining: IEmployeeTraining;
    //employeeIt: EmployeeIt;
    //employeeHr: EmployeeHr;

    projectName: string;
    jobSiteLocation: string;
    projectNumber: string;

    perDiem: string;
    perDiemTxt: string;

    mileageRate: string;
    mileageRateTxt: string;

    mileage: string;


    updatedByName: string;
    updatedByEmail: string;

    //createdByName: string;
    //createdByEmail: string;

}

export interface IEmployeeFormValues extends Partial<IEmployee> {
    time?: Date;
}

export class EmployeeFormValues implements IEmployeeFormValues {

    typeTechnical: string = '2';  // 1 = non-technical, 2 = technical, 3 = row

    didNotAcceptCbx: boolean = false;

    typeCbxNewHire: boolean = false;
    typeCbxReHire: boolean = false;
    typeCbxETChange: boolean = false;
    typeCbxUnion: boolean = false;
    typeCbxBManager: boolean = false;
    typeCbxTransfer: boolean = false;

    gotCbxMNFStandards: boolean = true;
    gotCbxTEExpenses: boolean = true;
    gotCbxMInternet: boolean = true;

    firstName: string = '';
    lastName: string = '';
    number: string = '';         // employee number

    startDate?: Date = undefined;
    startTime?: Date = undefined;

    office: string = '';
    location: string = '';
    pwLocation: string = '';
    employeeType: string = '';

    discipline: string = '';
    //discipline?: number;

    sF330Discipline: string = '';
    //sF330Discipline?: number;
    billingTitle: string = '';
    jobTitle: string = '';

    designation: string = '0';
    //designation: number = 0;

    managerName: string = '';
    managerEmail: string = '';
    companyCode: string = '';

    adminAssistant: string = '';
    ssNeeded: string = '';
    shNeeded: string = '';
    seatingArrangement: string = '';

    typeCbxMPosition: boolean = false;
    typeCbxESAMTraining: boolean = false;

    cvCbxEDMVehicle: boolean = false;
    cvCbxEAMVehicle: boolean = false;
    cvCbxGasCard: boolean = false;
    cvCbxEZPass: boolean = false;

    e52Week: string = '';
    //e52Week?: number;
    e48Week: string = '';
    //e48Week?: number;

    inCbxBCards: boolean = false;
    inCbxSPhone: boolean = false;
    inCbxPResume: boolean = false;
    inCbxLaptop: boolean = false;           // Office Laptop
    inCbxFieldLaptop: boolean = false;
    inCbxDesktop: boolean = false;
    inCbxUsbAccess: boolean = false;
    inCbxDPhone: boolean = false;
    inCbxOther: boolean = false;
    inTxtOther: string = '';

    inCbxMeetWithMarketing: boolean = false;

    seCbxHHats: boolean = false;
    seCbxSVests: boolean = false;
    seCbxSGlasses: boolean = false;
    seCbxOther: boolean = false;

    seHatColor: string = '';
    seNeededType: string = '';
    seSize: string = '';
    seTxtOther: string = '';

    employeeTraining: IEmployeeTraining = new EmployeeTrainingFormValues();
    employeeIt: IEmployeeIt = new EmployeeItFormValues();
    employeeHr: IEmployeeHr = new EmployeeHrFormValues();

    clientName: string = '';
    projectName: string = '';
    jobSiteLocation: string = '';
    projectNumber: string = '';

    perDiem: string = '';
    perDiemTxt: string = '';

    mileageRate: string = '';
    mileageRateTxt: string = '';

    cellPhone: string = '';
    computer: string = '';
    mileage: string = '';

    comments: string = '';

    updatedByName: string = '';
    updatedByEmail: string = '';

    //createdByName: string = '';
    //createdByEmail: string = '';

    constructor(init?: IEmployeeFormValues) {
        if (init && init.startDate) {
            init.startTime = init.startDate
        }

        Object.assign(this,init);
    }
}

File: Domain/Employee.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public class Employee
    {
        [Key]
        public string Number { get; set; }

        //[Required]
        //[StringLength(40)]
        public string FirstName { get; set; }

        //[Required]
        //[StringLength(40)]
        public string LastName { get; set; }

        public DateTime? StartDate { get; set; }

        //[Required]
        public Int16 Office { get; set; }

        //[Required]
        public Int16 Location { get; set; }

        //[StringLength(40)]
        public string PWLocation { get; set; }

        public Int16 EmployeeType { get; set; }
        public Int16  Discipline { get; set; }
        public Int16  SF330Discipline { get; set; }
        public Int16  BillingTitle { get; set; }

        //[StringLength(40)]
        public string JobTitle { get; set; }
        public Int16  Designation { get; set; }

        //[StringLength(40)]
        public string ManagerName { get; set; }

        //[StringLength(60)]
        public string ManagerEmail { get; set; }
        public Int16  CompanyCode { get; set; }

        //[StringLength(40)]
        public string AdminAssistant { get; set; }

        //[StringLength(200)]
        public string SSNeeded { get; set; }

        //[StringLength(200)]
        public string SHNeeded { get; set; }

        //[StringLength(200)]
        public string SeatingArrangement { get; set; }

        //[Required]
        //[StringLength(20)]
        public string TypeTechnical { get; set; }
        
        public Boolean DidNotAcceptCbx { get; set; }

        public Boolean TypeCbxMPosition { get; set; }
        public Boolean TypeCbxUnion { get; set; }
        public Boolean TypeCbxBManager { get; set; }
        public Boolean TypeCbxNewHire { get; set; }
        public Boolean TypeCbxReHire { get; set; }
        public Boolean TypeCbxETChange { get; set; }
        public Boolean TypeCbxESAMTraining { get; set; }
        public Boolean TypeCbxTransfer { get; set; }

        public Boolean GOTCbxMNFStandards { get; set; }
        public Boolean GOTCbxTEExpenses { get; set; }
        public Boolean GOTCbxMInternet { get; set; }

        public Boolean CVCbxEDMVehicle { get; set; }
        public Boolean CVCbxEAMVehicle { get; set; }
        public Boolean CVCbxGasCard { get; set; }
        public Boolean cvCbxEZPass { get; set; }

        public Int16 E52Week { get; set; }
        public Int16 E48Week { get; set; }

        public Boolean INCbxBCards { get; set; }
        public Boolean INCbxSPhone { get; set; }
        public Boolean INCbxPResume { get; set; }
        public Boolean INCbxLaptop { get; set; }        // Office Laptop

        public Boolean INCbxFieldLaptop { get; set; }

        public Boolean INCbxDesktop { get; set; }
        public Boolean INCbxUsbAccess { get; set; }
        public Boolean INCbxDPhone { get; set; }
        public Boolean INCbxOther { get; set; }

        //[StringLength(40)]
        public string INTxtOther { get; set; }

        public Boolean INCbxMeetWithMarketing { get; set; }

        public Boolean SECbxHHats { get; set; }
        public Boolean SECbxSVests { get; set; }
        public Boolean SECbxSGlasses { get; set; }
        public Boolean SECbxOther { get; set; }

        public Int16 SEHatColor { get; set; }
        public Int16 SENeededType { get; set; }
        public Int16 SESize { get; set; }

        //[StringLength(40)]
        public string SETxtOther { get; set; }

        public string EmployeeNumber { get; set; }

        public EmployeeTraining EmployeeTraining { get; set; }
        public EmployeeIt EmployeeIt { get; set; }
        public EmployeeHr EmployeeHr { get; set; }


        //[StringLength(40)]
        public string ClientName { get; set; }

        //[StringLength(40)]
        public string ProjectName { get; set; }

        //[StringLength(40)]
        public string JobSiteLocation { get; set; }

        //[StringLength(40)]
        public string ProjectNumber { get; set; }

        public Int16 PerDiem { get; set; }

        //[StringLength(40)]
        public string PerDiemTxt { get; set; }

        //[StringLength(40)]
        public Int16 MileageRate { get; set; }

        public string MileageRateTxt { get; set; }

        public Int16 CellPhone { get; set; }
        public Int16 Computer { get; set; }
        public Int16 Mileage { get; set; }

        //[StringLength(200)]
        public string Comments { get; set; }

        //[StringLength(40)]
        public string UpdatedByName { get; set; }

        //[StringLength(60)]
        public string UpdatedByEmail { get; set; }

        public DateTime? UpdateDate { get; set; }

        //[StringLength(40)]
        public string CreatedByName { get; set; }

        //[StringLength(60)]
        public string CreatedByEmail { get; set; }

        public DateTime CreateDate { get; set; }
         
    }
}